### ISSUE TITLE
Cannot validate second captcha after validation of the first one without refreshing the page.

### ISSUE THREAD

**Issue by Tookuk** [wbstr/vaadin-recaptcha#11]
Hi everybody!

I have an issue using the addon. 

What I do:
- I create ReCaptcha component and add it to my vaadin application page.
- I validate it, everything works fine.
- I recreate ReCaptcha component and add it to my page again, because some of the user actions need to be validated with captcha every time.
- Now there is no way to make captcha valid. It stays invalid until I refresh the browser page.

I tried to use reload() method, it didn't help.

Could you help me? Why is it happening?

**Comment by PAX523** [comment]
I've got the same issue. You can invoke ``recaptcha. validate()`` one time. But for all subsequent invocations it fails with following message:

`{
  "success": false,
  "challenge_ts": "2017-05-21T09:56:41Z",
  "hostname": "localhost",
  "error-codes": [
    "timeout-or-duplicate"
  ]
}`

With this behavior, this plugin becomes unusable in my context.

**Comment by plizio** [comment]
Same problem, fixed it!

The problem is in recaptcha-connector.js: at line 37 there is a call to recaptcha API function getResponse() w/o parameter.
This function accepts a parameter that is used to compose the id of recaptcha textarea that will contain the response.
The textarea changes id each validation, the first time is 'g-recaptcha-response', the second time is 'g-recaptcha-response-1' and so on.
The function getResponse() generates the id 'g-recaptcha-response' if no parameter or 0 is passed, generates id 'g-recaptcha-response-N' where N is the parameter if one is passed.
The call w/o parameter won't let validate after 1st time because the generated id will always be 'g-recaptcha-response' and the id of textarea will be 'g-recaptcha-response-N'.
The solution is to pass the parameter, my solution (that works but probably is not the best one) is to replace line 37 of file recaptcha-connector.js ( connector.responseChanged(grecaptcha.getResponse()); ) whit this code:

```
var cli = window.___grecaptcha_cfg.clients;
var num = 0;
if(cli != null && cli != undefined) {
	while(cli[num] != undefined)
		num++;
	num--;
}
connector.responseChanged(grecaptcha.getResponse(num));
```

where I get the object containing references to responses, I count them and pass the count to the function getResponse(). In this way I can retrieve the last response from google recaptcha APIs.

Hope this helps you.

**Comment by kumm** [comment]
Committed some fix.
Can you give it a try?

**Comment by plizio** [comment]
Perfect! Now it works correctly. Thanks kumm!

**Comment by kumm** [comment]
fix released in 2.0.3
